test/corpus/interfaces.txt

```diff
@@ -129,34 +128,31 @@ public interface IGetNext<T> where T : IGetNext<T>
         name: (identifier))
       (type_parameter
         name: (identifier)))
-    bases: (base_list
+    (base_clause
       (identifier))
     (type_parameter_constraints_clause
-      target: (identifier)
-      constraints: (type_parameter_constraint
-        (type_constraint
-          type: (identifier))))
+      (identifier)
+      (type_parameter_constraint
+        type: (identifier)))
     (type_parameter_constraints_clause
-      target: (identifier)
-      constraints: (type_parameter_constraint
+      (identifier)
+      (type_parameter_constraint
         (constructor_constraint)))
     body: (declaration_list))
   (namespace_declaration
     name: (identifier)
     body: (declaration_list
       (interface_declaration
         name: (identifier)
-        bases: (base_list
+        (base_clause
           (identifier))
         body: (declaration_list
-          (event_field_declaration
-            (variable_declaration
-              type: (generic_name
-                name: (identifier)
-                type_arguments: (type_argument_list
-                  (identifier)))
-              (variable_declarator
-                name: (identifier))))
+          (event_declaration
```

Comment:
Again this has gone from a definitive Roslyn-defined set to what appears to be an arbitrary choice/level of abstraction.

Here's the section from syntax analyzer:

![image](https://github.com/tree-sitter/tree-sitter-c-sharp/assets/118951/124cde95-ba1b-4902-8181-a06e613d3b67)

Reply:
no *this* one doesn't improve it all that much, to me I just thought that `event_declaration` satisfies this enough. this is innocuous enough to add back, but im curious why do we try to strictly follow the roslyn definition? I feel like that can cause some pain points in certain spots, but in this case it just adds a conflict that is sorta unnecessary, though state count only goes up a *smidge*

Reply:
Why would we not want to use the definitive name for the grammar parts rather than whoever adds a feature choose arbitrary names?